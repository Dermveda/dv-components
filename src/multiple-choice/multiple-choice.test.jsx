import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MultipleChoice from './multiple-choice.jsx';
const question = {
	_id: 'abc123',
	name: 'test q',
	type: 'test',
	isCorrect: true,
	chosenAnswer: {
		_id: 'abc234',
		name: 'test a',
		isChosen: true,
		isCorrect: false
	},
	answers: [
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc235',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc236',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc237',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc238',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		}
	]
};

describe('<MultipleChoice/>', () => {
	it('Will render', () => {
		const el = renderer.create(<MultipleChoice question={question} />);
		expect(el).toMatchSnapshot();
	});
	it('Will have question', () => {
		const el = shallow(<MultipleChoice question={question} />);
		expect(el).toMatchSnapshot();
	});
	it('Will select answer', () => {
		const func = jest.fn();
		const el = mount(<MultipleChoice question={question} onSelectAnswer={func} />);
		el
			.find('div.dvc-card')
			.first()
			.simulate('click');
		expect(func).toHaveBeenCalledTimes(1);
	});
});
