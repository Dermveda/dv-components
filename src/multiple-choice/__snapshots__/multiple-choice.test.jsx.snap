// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleChoice/> Will have question 1`] = `
<li>
  <div
    className="pb-1"
  >
    test q
  </div>
  <div>
    <div
      className="my-0 dvc-card dvc-card--enabledMinimal"
      key="QuizQuestionAnswer_abc123_abc234"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 dvc-primary-text align-middle"
        >
          check_circle
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      key="QuizQuestionAnswer_abc123_abc235"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      key="QuizQuestionAnswer_abc123_abc236"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      key="QuizQuestionAnswer_abc123_abc237"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      key="QuizQuestionAnswer_abc123_abc238"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
  </div>
</li>
`;

exports[`<MultipleChoice/> Will render 1`] = `
<li>
  <div
    className="pb-1"
  >
    test q
  </div>
  <div>
    <div
      className="my-0 dvc-card dvc-card--enabledMinimal"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 dvc-primary-text align-middle"
        >
          check_circle
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
    <div
      className="my-0 dvc-card pointer dvc-card--disabledMinimal"
      onClick={[Function]}
    >
      <div
        className="my-auto"
      >
        <i
          className="material-icons pr-2 align-middle"
        >
          radio_button_unchecked
        </i>
        test a
      </div>
    </div>
  </div>
</li>
`;
